include/modem/lte_lc.h

```diff
@@ -2095,6 +2303,159 @@ int lte_lc_periodic_search_clear(void);
+ */
+int lte_lc_context_auth_set(uint8_t cid, enum lte_lc_pdn_auth method,
+			    const char *user, const char *password);
+
+/**
+ * Destroy a PDP context.
+ *
+ * The PDN connection must be inactive when the PDP context is destroyed.
+ *
+ * @param[in] cid The context ID.
+ *
+ * @retval 0 On success.
+ * @return A negative errno otherwise.
+ */
+int lte_lc_context_destroy(uint8_t cid);
+
+/**
+ * Activate a PDN connection.
+ *
+ * Activates the PDP context with the given CID, establishing a PDN connection.
+ *
+ * @param[in] cid The context ID.
+ * @param[out] esm If provided, the function will block to return the ESM error reason.
+ * @param[out] family If provided, the function will block to return
+ *		      @c LTE_LC_PDN_FAM_IPV4 if only IPv4 is supported, or
+ *		      @c LTE_LC_PDN_FAM_IPV6 if only IPv6 is supported.
+ *		      Otherwise, this value will remain unchanged.
+ *
+ * @retval 0 On success.
+ * @return A negative errno otherwise.
+ */
+int lte_lc_context_activate(uint8_t cid, int *esm, enum lte_lc_pdn_family *family);
+
+/**
+ * Deactivate a PDN connection.
+ *
+ * Deactivates the PDN connection associated with the given context ID.
+ *
+ * @param[in] cid The context ID.
+ *
+ * @retval 0 On success.
+ * @return A negative errno otherwise.
+ */
+int lte_lc_context_deactivate(uint8_t cid);
+
+/**
+ * Retrieve the PDN ID for a given PDP context.
+ *
+ * The PDN ID can be used to route traffic through a PDN connection.
+ * Multiple contexts (CIDs) may share the same PDN ID if they route through
+ * the same network connection.
+ *
+ * @param[in] cid The context ID.
+ *
+ * @return A non-negative PDN ID on success.
+ * @return A negative errno otherwise.
+ */
+int lte_lc_context_pdn_id_get(uint8_t cid);
+
+/**
+ * Retrieve dynamic parameters of a given PDN connection.
+ *
+ * @param[in] cid The context ID.
+ * @param[out] info PDN dynamic info.
+ *
+ * @retval 0 On success.
+ * @return A negative errno otherwise.
+ */
+int lte_lc_context_pdn_dynamic_info_get(uint8_t cid, struct lte_lc_pdn_dynamic_info *info);
+
+/**
+ * Retrieve the default Access Point Name (APN).
+ *
+ * The default APN is the APN of the default PDP context (zero).
+ *
+ * @param[out] buf The buffer to copy the APN into. The string is null-terminated.
+ * @param[in] len The size of the output buffer.
+ *
+ * @retval 0 On success.
+ * @return A negative errno otherwise.
+ */
+int lte_lc_default_apn_get(char *buf, size_t len);
+
+/**
+ * Configure a cellular profile.
+ *
+ * The cellular profile to be used for a specific access technology is decided by the modem.
+ * The active cellular profile is reported in the @ref LTE_LC_EVT_CELLULAR_PROFILE_ACTIVE
+ * event if an event handler is registered.
+ *
+ * @param[in] profile The cellular profile to set.
+ *
+ * @retval 0 on success, otherwise negative error code.
+ */
+int lte_lc_cellular_profile_configure(struct lte_lc_cellular_profile *profile);
+
+#if CONFIG_LTE_LC_PDN_ESM_STRERROR
```

Comment:
`#if defined` for Kconfigs, never just `#if` them, plus this prevents any documentation, so check for `defined(__DOXYGEN__)` too

Reply:
Fixed, added a bunch of `defined(__DOXYGEN__)` in header file as well to fix doc build